src/mono/mono/metadata/domain-internals.h

```diff
@@ -720,6 +717,23 @@ mono_domain_ambient_alc (MonoDomain *domain)
 	return mono_domain_default_alc (domain);
 }
 
+static inline MonoMemoryManager *
+mono_domain_default_memory_manager (MonoDomain *domain)
+{
+#ifdef ENABLE_NETCORE
+	return (MonoMemoryManager *)mono_domain_default_alc (domain)->memory_manager;
+#else
+	return domain->memory_manager;
+#endif
+}
+
+static inline MonoMemoryManager *
+mono_domain_ambient_memory_manager (MonoDomain *domain)
+{
+	// FIXME: All callers of mono_domain_ambient_memory_manager should get a MemoryManager from their callers or context
```

Comment:
So this is used to mark locations which need to be changed ? In that case, can mono_domain_default_memory_manager () be named simply mono_domain_memory_manager () ?

Reply:
Sure.

Reply:
And yes, in most of those cases we want to get the MemoryManager from a class or method. This PR doesn't include the generics stuff, so I'm just putting this for now. There will be more of it when removing `mono_domain_alloc` calls.